compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1986,6 +1987,42 @@ impl CheckAttrVisitor<'_> {
         }
     }
 
+    fn check_confusables(&self, attr: &Attribute, target: Target) -> bool {
+        match target {
+            Target::Method(MethodKind::Inherent) => {
+                let Some(meta) = attr.meta() else {
+                    self.tcx.sess.emit_err(errors::FailedToExtractMetaItem { span: attr.span });
+                    return false;
+                };
+                let ast::MetaItem { kind: MetaItemKind::List(ref metas), .. } = meta else {
+                    self.tcx.sess.emit_err(errors::InvalidMetaItemKind { span: attr.span });
+                    return false;
+                };
```

Comment:
I don't think you need to emit an error in those cases, just early return. You already get a parser error.